doc/site/docs/02-reference/01-handlers.md

```diff
@@ -12,38 +12,30 @@ What makes Relic handlers special is their flexibility. Unlike traditional web f
 - Upgrade connections to WebSockets for real-time communication
 - Take direct control of the network connection for custom protocols
 
-This is why Relic handlers work with "context" objects instead of just taking a request and returning a response. The context gives you multiple ways to handle the connection based on what your application needs.
+Relic handlers take a `Request` and return a `Result`, which can be a `Response`, `Hijack` (for connection hijacking), or `WebSocketUpgrade` (for WebSocket connections).
 
 ## Handler types
 
-In Relic, there are four main handler types for different scenarios:
+In Relic, there are two main handler types for different scenarios:
 
 ### 1. Handler (foundational)
 
-The `Handler` is the foundational handler type that serves as the core building block for all Relic applications. It provides the most flexibility by accepting any incoming request context and allowing you to return various types of handled contexts, making it suitable for any kind of request processing logic.
+The `Handler` is the foundational handler type that serves as the core building block for all Relic applications. It provides the most flexibility by accepting any incoming request and allowing you to return various types of results, making it suitable for any kind of request processing logic.
 
 ```dart
-typedef Handler = FutureOr<HandledContext> Function(NewContext ctx);
+typedef Handler = FutureOr<Result> Function(Request req);
 ```
 
-- `NewContext ctx` is the incoming request context.
-- `FutureOr<HandledContext>` is a response context, hijacked connection, or WebSocket connection.
+- `Request req` is the incoming HTTP request.
+- `FutureOr<Result>` is a result that can be a `Response`, `Hijack`, or `WebSocketUpgrade`.
 
 **Example:**
 
-```dart
-import 'package:relic/relic.dart';
-
-Handler helloHandler(NewContext ctx) {
-  return ctx.respond(Response.ok(
-    body: Body.fromString('Hello from Relic!'),
-  ));
-};
-```
+GITHUB_CODE_BLOCK lang="dart" doctag="handler-foundational" [src](https://raw.githubusercontent.com/serverpod/relic/main/example/basic/handler_example.dart) title="Foundational Handler example"
 
 ### 2. Responder
 
-A `Responder` is a simplified function type that provides a more straightforward approach to request handling by directly transforming an HTTP request into an HTTP response. It abstracts away the context handling, making it perfect for simple request-response scenarios and can easily be converted into a full `Handler` using the `respondWith` helper function.
+A `Responder` is a simplified function type that provides a more straightforward approach to request handling by directly transforming an HTTP request into an HTTP response. It can easily be converted into a full `Handler` using the `respondWith` helper function.
```

Comment:
We should get rid of `Responder` as well. It doesn't really add anything any more.